nornir/plugins/inventory/netbox.py

```diff
@@ -37,6 +37,17 @@ def __init__(
             if d.get("primary_ip", {}):
                 temp["nornir_host"] = d["primary_ip"]["address"].split("/")[0]
 
+                # Find interface name using additional API call
+                i = requests.get(
+                    "{}/api/ipam/ip-addresses/?address={}".format(
+                        nb_url, temp["nornir_host"]
+                    ),
+                    headers=headers,
+                ).json()
+
```

Comment:
By default, `requests` does not raise an exception when something goes wrong.
Could you please add something like this:
```
response = requests.get(...)
if response.ok:
    temp["primary_interface"] = response.json()...
```
Alternatively, you can achieve similar thing using `response.raise_for_status()` and catching `requests` exceptions.

Reply:
I'd suggest using `raise_for_status` as above.

Reply:
Technically, the first `r.raise_for_status()` line [27](https://github.com/nornir-automation/nornir/pull/222/files#diff-71d4c60e78b698e99f071d4074c15052R27) will catch any errors in the submitted request (i.e. an interface **MUST** have an address first, only if that is successful, then I'm querying for the name).

The only way a second error would occur is if the entire DB is screwed up.
Then if that is the case, you have some other issues to worry about.

All that being said, the likelihood that this explanation is going to be remembered in even 1 months time-frame is small, so I'm okay adding the additional `r.raise_for_status()` simply for documentation purposes.

However, I don't have the time this week to add data for the `requests.mock` tests that are failing, so I'll do this next week so as to not have several failed `travis-ci` tests on record :smile:

Reply:
> The only way a second error would occur is if the entire DB is screwed up

There could be other reasons, load causing a timeout, lots of connections opened against the databases leading to starvation of resources, a service crash... We should check that on every request just for safety.

> I don't have the time this week to add data for the requests.mock tests that are failing

No rush, anything you need let us know.

Reply:
As I said in the comment on the recent commit I made, I don't know how to integrate the `JSON` files for the `address_[1,2,3]` in your test file which is why test 1 (`py36`) is failing.

And after spending _some_ learning about what `mypy` is, I have no idea why test 5 is failing.